chia/protocols/protocol_state_machine.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional
+
+from chia.protocols.protocol_message_types import ProtocolMessageTypes as pmt, ProtocolMessageTypes
+
+NO_REPLY_EXPECTED = [
+    # full_node -> full_node messages
+    pmt.new_peak,
+    pmt.new_transaction,
+    pmt.new_unfinished_block,
+    pmt.new_signage_point_or_end_of_sub_slot,
+    pmt.request_mempool_transactions,
+    pmt.new_compact_vdf,
+    pmt.request_mempool_transactions,
+]
+
+"""
+VAILD_REPLY_MESSAGE_MAP:
+key: sent message type.
+value: valid reply message types, from the view of the requester.
+A state machine can be built from this message map.
+"""
+
+VAILD_REPLY_MESSAGE_MAP = {
+    # messages for all services
+    # pmt.handshake is handled in WSChiaConnection.perform_handshake
+    # full_node -> full_node protocol messages
+    pmt.request_transaction: [pmt.respond_transaction],
+    pmt.request_proof_of_weight: [pmt.respond_proof_of_weight],
+    pmt.request_block: [pmt.respond_block, pmt.reject_block],
+    pmt.request_blocks: [pmt.respond_blocks, pmt.reject_blocks],
+    pmt.request_unfinished_block: [pmt.respond_unfinished_block],
+    pmt.request_signage_point_or_end_of_sub_slot: [pmt.respond_signage_point, pmt.respond_end_of_sub_slot],
+    pmt.request_compact_vdf: [pmt.respond_compact_vdf],
+    pmt.request_peers: [pmt.respond_peers],
+}
+
+
+def static_check_sent_message_response() -> None:
```

Comment:
I would even think this check would make sense to put in the global scope, to fail immediately when this module is `import`ed, if we have this error.

Reply:
Yes, good point. That's more "static" !